packages/flutter/lib/src/material/divider.dart

```diff
@@ -117,9 +117,21 @@ class Divider extends StatelessWidget {
   /// ```
   /// {@end-tool}
   static BorderSide createBorderSide(BuildContext context, { Color color, double width }) {
+    final Color themeColor = context != null ? (DividerTheme.of(context).color ?? Theme.of(context).dividerColor) : null;
+    final Color effectiveColor = color ?? themeColor;
```

Comment:
I think this should be 
```
final Color effectiveColor = color ?? (context != null ? DividerTheme.of(context).color : null);
```
DividerTheme.of() already defers to Theme.of() and if color is specified there's no need to create an (inherited widget) dependency on a theme.

Reply:
Unfortunately, this wouldn't allow us to use Theme.of().dividerColor as a fallback (the legacy behavior), so I'd have to keep that

Reply:
Good point. So long as it's all part of an expression that short-circuits if color is non-null.

Reply:
but I consolidated it into one variable

Reply:
Thanks, looks good.